implementation/create_name_procedure.sql

```diff
@@ -0,0 +1,205 @@
+    @genative_pronouns AS VARCHAR,                      -- Name Prefix - logically grouped with gender information
+    @honorific_text As VARCHAR, 
+
+    -- ****** CULTERAL IDENTITY INFORMATION (see UI mockup) ****** --
+    @tribe_clan_name_unicode  AS NVARCHAR,               -- text of tribe or clan unicode name enered into the UI.
+    @tribe_clan_name_latin1 as VARCHAR,                 -- text of tribe or clan unicode name enered into the UI.
+    @tribe_clan_name_ipa AS NVARCHAR,                   -- text of tribe or clan unicode name enered into the UI.
+
+    @mother_person_id as INT,                           -- passed if association is made on the name record, null if not
+    @father_person_id as INT,                           -- passed if association is made on the name record, null if not
+
+    -- ****** OVERRIDE IDENTITY INFORMATION (See UI mockup) ****** --
+    @override_full_name_unicode as NVARCHAR,           -- text of unicode full override name entered into the UI.
+    @override_full_name_latin1 as VARCHAR,             -- text of latin1 full override name entered into the UI.
+    @override_full_name_ipa AS NVARCHAR,               -- text for ipa of full override name value entered into the UI.
+
+)
+AS BEGIN
+    BEGIN TRANSACTION
+        BEGIN TRY
+            -- Orchestrate proper order of operations for inserting a name
+
+            -- Step 1: Call f_validate_email and throw error if email already in the database
+                -- if there is a match, throw an error
+                -- Piggy: No, if the email is already in the database, get the person_id that goes with it and use that going forward.
+                --        It is perfectly reasonable to create more than one name for a given person.
+
+                -- if no match, perform an insert into the persons table and capture new ID in a variable
+                -- variables created: @new_person_id
+            
+            -- Piggy: Rather than these validation functions, provide conditional insert functions that return the id.
+            --- If the particle exists, just return the id, otherwise create the entry and return the id.
+
+            -- Step 2: Call a f_validate_name_id function (unicode, latin1, ipa) to confirm if entry already in database. 
+                -- If function returns null, proceed with insert and capture new ID in a variable
+                -- If function returns an ID, simply store the ID in a new variable.
+                -- repeat this process for all submitted name information that maps to unique particles
+
+                -- Piggy: I recommend not checking the latin1 or ipa components of the record. All three forms should
+                --    should resolve to the same id. If we didn't specify a uniqueness constraint on (unicode, type),
+                --    we should have.
+                -- We should only create one id per triple. It's OK to check that the latin1 and ipa match, but

```

Comment:
Alternatively, we can always update the values for latin1 and ipa. That means the create function and the update function are one and the same.